### PR TITLE
Feature/journey

### PR DESCRIPTION
**Pull request by geralddejong**

### CODE REVIEW

File: client/src/app-storage.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 
 const MASTER_KEY = "master"
 
+function journeyKey(hexalot: Hexalot): string {
+    return `${hexalot.createFingerprint()}-journey`
+}
+
+function genomeKey(hexalot: Hexalot): string {
+    return hexalot.createFingerprint()
+    // return `${hexalot.createFingerprint()}-genome`
```
Comment (mitchelljustin) [inline client/src/app-storage.ts]: Remove

File: client/src/app.tsx [hunk 2]
```diff
@@ -58,15 +65,15 @@ function startEvolution(hexalot: Hexalot): object {
     return (state: IAppState, props: IAppProps) => {
         state.island.setIslandState(true, hexalot)
         dispose(state)
-        const trip = hexalot.createStupidTrip()
+        const journey = hexalot.createStupidJourney()
```
Comment (mitchelljustin) [inline client/src/app.tsx:68]: Lol

File: client/src/app.tsx [hunk 3]
```diff
@@ -111,10 +118,10 @@ function setInfoPanelMaximized(maximized: boolean): void {
 
 class App extends React.Component<IAppProps, IAppState> {
     private subs: Subscription[] = []
-    private orbitDistanceSubject: BehaviorSubject<OrbitDistance> =
-        new BehaviorSubject<OrbitDistance>(OrbitDistance.HELICOPTER)
     private perspectiveCamera: PerspectiveCamera
-    private selectedSpotSubject: BehaviorSubject<Spot | undefined> = new BehaviorSubject<Spot | undefined>(undefined)
+    private hexalotIdSubject = new BehaviorSubject<string>("")
+    private orbitDistanceSubject = new BehaviorSubject<OrbitDistance>(OrbitDistance.HELICOPTER)
+    private selectedSpotSubject = new BehaviorSubject<Spot | undefined>(undefined)
```
Comment (mitchelljustin) [inline client/src/app.tsx:124]: You don't have to pass `undefined`

File: client/src/app.tsx [hunk 4]
```diff
@@ -147,12 +154,14 @@ class App extends React.Component<IAppProps, IAppState> {
...
             height: window.innerHeight,
+            left: window.screenLeft,
+            top: window.screenTop,
         }
         this.perspectiveCamera = new PerspectiveCamera(50, this.state.width / this.state.height, 1, 500000)
     }
 
     public componentDidMount(): void {
-        window.addEventListener("resize", () => this.setState(updateDimensions))
+        window.addEventListener("resize", (e) => this.setState(updateDimensions))
```
Comment (mitchelljustin) [inline client/src/app.tsx]: Don't need the event here?

File: client/src/app.tsx (hunk 3 above)
Comment (geralddejong) [reply client/src/app.tsx:124]: Red underline in IntelliJ if I don't.

File: client/src/app.tsx [hunk 5]
```diff
@@ -167,6 +176,38 @@ class App extends React.Component<IAppProps, IAppState> {
                 this.fabricKernel.setHexalot(spotCenters, surface)
             }
         }))
+        this.subs.push(this.hexalotIdSubject.subscribe(hexalotId => location.replace(`/#/${hexalotId}`)))
+        this.subs.push(this.selectedSpotSubject.subscribe(spot => {
+            if (spot) {
```
Comment (mitchelljustin) [inline client/src/app.tsx]: This is nicer to read if you instead do 
```
            if (!spot) {
              return
            }
            const hexalot = spot.centerOfHexalot
            if (!hexalot) {
              return
            }
            ...
```

**Review by mitchelljustin — APPROVED** [review]
Code style

File: client/src/app.tsx (hunk 3 above)
Comment (mitchelljustin) [reply client/src/app.tsx:124]: Ah, ok

**Pull request merged**